src/Node/ClassPropertiesNode.php

```diff
@@ -98,10 +103,7 @@ public function getUninitializedProperties(
 		if (!$this->getClass() instanceof Class_) {
 			return [[], [], []];
 		}
-		if (!$scope->isInClass()) {
-			throw new ShouldNotHappenException();
-		}
-		$classReflection = $scope->getClassReflection();
+		$classReflection = $this->getClassReflection();
```

Comment:
the `$scope` param on this method is no longer used as we can access the class reflection on the node now, however removing it would be a BC break and this class is covered by `@api` so i just left it there